server/src/test/java/org/elasticsearch/index/shard/IndexShardOperationPermitsTests.java

```diff
@@ -220,17 +231,23 @@ public void testGetBlockWhenBlocked() throws ExecutionException, InterruptedExce
         try (Releasable ignored = blockAndWait()) {
             permits.acquire(future, ThreadPool.Names.GENERIC, true, "");
 
-            permits.asyncBlockOperations(
-                30,
-                TimeUnit.MINUTES,
-                () -> {
-                    blocked.set(true);
-                    blockAcquired.countDown();
-                    releaseBlock.await();
-                },
-                e -> {
+            permits.asyncBlockOperations(new ActionListener<Releasable>() {
+                @Override
+                public void onResponse(final Releasable releasable) {
```

Comment:
we use this pattern often  - maybe introduce a simple class that wraps a runnable and runs it while releasing the releasable?